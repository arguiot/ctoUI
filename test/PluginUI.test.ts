import { JSDOM } from "jsdom";
import {
  Algorithm,
  PluginUI,
  Configuration,
  Text,
  Direction,
} from "../dist/index";

describe("Render Input", () => {
  it("should render Input", () => {
    class Algo implements Algorithm {
      bindings: Configuration;
      constructor(bindings: Configuration) {
        this.bindings = bindings;
      }
      encode() {
        return;
      }
    }
    const { window } = new JSDOM();

    const plugin = new PluginUI(Algo, {
      input: Text,
      output: Text,
      direction: Direction.Unidirectional,
      options: {},
      parent: window.document.body,
    });

    plugin.render(window);

    expect(window.document.body.innerHTML).toMatchSnapshot()
  });
});
