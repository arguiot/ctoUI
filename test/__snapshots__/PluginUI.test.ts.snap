// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Input should render Input 1`] = `"<div class=\\"input\\"><div class=\\"form-group mt-3\\"><label>Label</label><textarea rows=\\"4\\" class=\\"form-control\\">Input</textarea><span id=\\"field-len\\" class=\\"float-right length-label\\">length: 5</span></div></div><div class=\\"output\\"><div class=\\"form-group mt-3\\"><label>Label</label><textarea rows=\\"4\\" class=\\"form-control\\">Input</textarea><span id=\\"field-len\\" class=\\"float-right length-label\\">length: 5</span></div></div>"`;
